fab_test.py

```diff
@@ -263,33 +271,30 @@ def update_build_poc(conns):
             )
 
 
-exe_name = "sharding-poc"
+exe_name = "sharding-p2p-poc"
 
 
 def run_servers(conns):
     commands = {}
     for node_index, _ in enumerate(conns):
+
+        linux_command = f"script -f -c './{exe_name} -seed={node_index}' poc_{node_index}.out"
+        osx_command = f"script -q /dev/null ./{exe_name} -seed={node_index} 2>&1 1>poc_{node_index}.out /dev/null"
+
+
         program_cmd = make_batch_cmd([
             f"killall -9 {exe_name}",
-            "sleep 1",
-            f"nohup script -c './sharding-poc -seed={node_index} 2>&1 1>poc_{node_index}.out &' /dev/null",
+             "sleep 1",
+             f"screen -d -m bash -c \"if [  \"$(uname)\" == \"Darwin\"  ]; then {osx_command}; else {linux_command}; fi\"",
         ])
 
         node_cmd = make_and_cmd([
             "cd {}".format(make_repo_src_path(poc_repo)),
             program_cmd,
         ])
-        # node_cmd = "nohup sleep 20 > 123.txt < /dev/null &"
-        # node_cmd = "cd {}".format(make_repo_src_path(poc_repo))
-        # node_cmd += f"&& nohup ./{exe_name} -seed={node_index} > poc_node_{node_index}.out < /dev/null & "
-        # node_cmd = "nohup ./sharding-poc -seed=0 /dev/null &"
-        # node_cmd = "(nohup ./sharding-poc -seed 0 &) && ps aux|grep sharding-poc"
-        # node_cmd = "setsid ./sharding-poc -seed 0"
-        # node_cmd = "python3 ./temp.py > haha.txt"
-        # node_cmd = "nohup ./c_sleep > haha.txt &"
         commands[node_index] = node_cmd
-    print(commands)
-    g = ThreadingGroup.from_connections(node_conns).run(custom_kwargs=commands)
+
+    g = ThreadingGroup.from_connections(node_conns).run(custom_kwargs=commands,echo=True)
```

Comment:
What is `echo=True` used for? The script seems to be working without that.

Reply:
Ya, you don't need echo=True. I used it bc it printed out the exact command that was being run. It was only useful for debugging.